Framework/Geometry/inc/MantidGeometry/Instrument/ParameterMap.h

```diff
@@ -380,6 +383,14 @@ class MANTID_GEOMETRY_DLL ParameterMap {
   /// associated with an ExperimentInfo object.
   std::unique_ptr<Geometry::ComponentInfo> m_componentInfo;
 
+  /// Pointer to the DetectorInfo wrapper. NULL unless the instrument is
+  /// associated with the instrument and that has a physical instrument.
+  std::unique_ptr<Geometry::DetectorInfo> m_detectorInfoPhysical;
```

Comment:
Ok, this brings us to the discussion we had yesterday: now that we have this, how do you make sure `Component::getPos` works correctly on the physical instrument? https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument/Component.cpp#L343.

Reply:
Yes. I see your point. As it stands, you would have to know that the component in question belonged to either the physical or neutronic instrument. The two options I see are.

1. We try to lookup the component (via ID) in the `ComponentInfo` and if it's not there for the neutronic one, we look for it in the physical one.
1. We set a flag on the Instrument to indicate that it is either a physical or neutronic one. We know this at the time that they are created. However, this would looking up the root (instrument object) every time you performed an operation.